### ISSUE TITLE
Bond Portfolio Object for Strategy Analysis

### ISSUE THREAD

**Issue by domokane** [domokane/FinancePy#32]
Calculate duration for portfolio of bonds

**Comment by quantophile** [comment]
Hi Dominic, is this issue this open? I am currently reading Schreve's book and am interested to actively contribute to FinancePy and strengthen my technical skills.

**Comment by domokane** [comment]
Yes. You're welcome to to do it. Let me know how it goes.
Best
D

**Comment by DominicHong** [comment]
Hi @domokane 
Is the product.bonds.bond_portfolio.py for this issue? It's not implemented yet. So the goal of this issue is to implement the bond_portfolio.py?

**Comment by domokane** [comment]
Exactly. If you wish to have a go, be my guest. Ultimate idea would be to have a portfolio object that you can use to analyse multi-bonds for hedging and scenario analysis and curve fitting.

**Comment by DominicHong** [comment]
I am thinking about a method of computing the profit of a Bond given the inital ytm and final ytm.
It should look like this:
```Python
def get_profit(self, initial_date, end_date, begin_ytm, end_ytm):
```
It adds all the cashflow during the period and calculate the simple return and compounded return.
Where should I put this method?

**Comment by sagayev** [comment]
I also would like to work on this. But before that someone with more strategic view should implement base bond class and other bonds should inherit from that base.
Base bond class should have at least the following attributes
"Currency" 
"Issue Date"
"Interest Accrues" - sometimes it can be later or before than issue date
"Maturity Date" 
"Floater Formula"
"Security identifier" -- at least ISIN

Then the Portfolio class should have a base currency. 
If I implement, for example,   
if not bondWeights:
    #calculate the sum of bond dirty prices
    #calculate weight of each bond
    #return ndarray bondWights

if in the future there will be multicurrency portfolio, the code will need to change again.

@domokane please let me know if should ignore these challenges and work on portfolio class right away.

**Comment by DominicHong** [comment]
Hi @sagayev,
product.bonds.Bond is the base class you want, except for the Security Identifier attribute. 
However, all types of bonds are implemented individually, not by inheritance. Domokane prefered this way for simplicity.
What's Floater Formula by the way?

**Comment by domokane** [comment]
Hi - I am not against refactoring the code to have a bond base class. I prefered to keep it separate for the specific case of a zero coupon bond. But if the benefits outweigh the cost then I am happy to reconsider.
